File: app/src/main/java/feelings/guide/profile/LocaleChangeListener.java
package feelings.guide.profile;

import android.content.Context;
import android.content.SharedPreferences;

import feelings.guide.R;
import feelings.guide.question.QuestionService;
import feelings.guide.util.ToastUtil;

public class LocaleChangeListener implements SharedPreferences.OnSharedPreferenceChangeListener {
    private Context context;

    public LocaleChangeListener(Context context) {
        this.context = context;
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        if (LocaleUtil.SELECTED_LANGUAGE.equals(key)) {
            context = LocaleUtil.setLocale(context);
            QuestionService.changeLanguage(context);
            ToastUtil.showLong(context, context.getString(R.string.msg_change_language_restart));
        }
    }
}

File: app/src/main/kotlin/feelings/guide/profile/LocaleChangeListener.kt
package feelings.guide.profile

import android.content.Context
import android.content.SharedPreferences

import feelings.guide.R
import feelings.guide.question.QuestionService
import feelings.guide.util.ToastUtil

class LocaleChangeListener(private var context: Context?) : SharedPreferences.OnSharedPreferenceChangeListener {

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        if (LocaleUtil.SELECTED_LANGUAGE == key) {
            context = LocaleUtil.setLocale(context!!)
            QuestionService.changeLanguage(context!!)
            ToastUtil.showLong(context!!, context!!.getString(R.string.msg_change_language_restart))
        }
    }
}
